nixos/modules/services/monitoring/prometheus/exporters/folder-size.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, options, utils }:
+let
+  cfg = config.services.prometheus.exporters.folder-size;
+
+  settingsFormat = pkgs.formats.json { };
+  settingsFile = settingsFormat.generate "config.json" (builtins.filter (folder: builtins.hasAttr "path" folder) cfg.folders);
+in
+{
+  port = 9974;
+  extraOpts = {
+    folders = lib.mkOption {
+      description = "Folders to Scan";
+      type = lib.types.listOf (lib.types.submodule {
+        options = {
+          path = lib.mkOption {
+            description = "The starting analysis path.";
+            type = lib.types.str;
+          };
+          explode_depth = lib.mkOption {
+            description = ''
+              This setting controls how deep the folder explosion will go.
+              -1 means no limit.
+              0 means no explosion.
+            '';
+            type = lib.types.int;
+            default = 0;
+          };
+          sum_remaining_subfolders = lib.mkOption {
+            description = ''This setting specifies if the last exploded folder size should include the subfolders.'';
+            type = lib.types.bool;
+            default = false;
+          };
+          user = lib.mkOption {
+            description = "Folders belonging to this user you want to specifically label.";
+            type = lib.types.nullOr lib.types.str;
+            default = null;
+          };
+        };
+      });
+    };
+
+    background = lib.mkOption {
+      type = lib.types.int;
+      default = 0;
+      description = ''
+        Enables the async storage calculation.
+        The value specifies how often (in seconds) the calculation will be done.
+        If not specified, the values will be calculated synchronously at each HTTP Get.
+      '';
+    };
+
+    verbose = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''Enable verbose logging of service.'';
+    };
+
+    directory = {
```

Comment:
This is where the config files and logging data goes.  
Possibly bad naming on my part.

Reply:
Yes, but is this option really necessary? Why would you use something other than `/var/lib/prometheus-folder-size-exporter`?

Reply:
Maybe not the ``folder`` but definitely the ``root``, since someone may want to have it store the info on a mounted drive.

Can remove the folder if ye want, dosent really matter too much

Reply:
> someone may want to have it store the info on a mounted drive

Yes, but why would they want to? Does prometheus-folder-size-exporter hog disk space? If not it's better to just drop this option. If someone really wants to, they could always use a bind mount.